### PR TITLE
[Feature]Broker oss support

### PR DESCRIPTION
**Pull request by gengjun-git**
Support aliyun oss read/write in broker, through hadoop-oss client.

### CODE REVIEW

**Review by francisoliverlee — COMMENTED** [review]
fe hard-coded to check the schema, oss should added here.  and i had checcked it by mine oss broker
`
https://github.com/apache/incubator-doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java
`
image

File: fs_brokers/apache_hdfs_broker/build.sh [hunk 1]
```diff
@@ -42,6 +42,12 @@ rm -rf ${BROKER_OUTPUT}
 install -d ${BROKER_OUTPUT}/bin ${BROKER_OUTPUT}/conf \
            ${BROKER_OUTPUT}lib/
 
+# download aliyun hadoop oss jar, this can not be found in maven repository, so we download it manually
+wget http://gosspublic.alicdn.com/hadoop-spark/hadoop-oss-2.7.2.tar.gz
+tar xzf hadoop-oss-2.7.2.tar.gz
+mv hadoop-oss-2.7.2/hadoop-aliyun-2.7.2.jar ${BROKER_OUTPUT}/lib/
+rm -r hadoop-oss-2.7.2 hadoop-oss-2.7.2.tar.gz
```
Comment (morningman) [inline fs_brokers/apache_hdfs_broker/build.sh:49]: Is it must be 2.7.2?
Does this not work?
`https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-aliyun/2.9.2`

Comment (gengjun-git) [reply fs_brokers/apache_hdfs_broker/build.sh:49]: Not work, I have tried the 2+ and 3+ version of lib from maven repository. So we have to download it by hand.

**Comment by gengjun-git** [comment]
> fe hard-coded to check the schema, oss should added here. and i had checcked it by mine oss broker
> `https://github.com/apache/incubator-doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java`
> image

Ok, I have changed and commited.

File: docs/zh-CN/sql-reference/sql-statements/Data Manipulation/BROKER LOAD.md [hunk 2]
```diff
@@ -477,6 +485,21 @@ under the License.
...
+    15. 从 Aliyun OSS 导入csv格式的数据
+        LOAD LABEL example_db.label15
+        (
+        DATA INFILE("oss://my_bucket/input/file.csv")
+        INTO TABLE `my_table`
+        (k1, k2, k3)
+        )
+        WITH BROKER my_oss_broker
+        (
+        "fs.oss.accessKeyId" = "xxxxxxxxxxxxxxxxxxxxxxxxxx",
```
Comment (wuyunfeng) [inline docs/zh-CN/sql-reference/sql-statements/Data Manipulation/BROKER LOAD.md:498]:
```suggestion
        "fs.oss.accessKey" = "xxxxxxxxxxxxxxxxxxxxxxxxxx",
```

File: docs/zh-CN/sql-reference/sql-statements/Data Manipulation/BROKER LOAD.md [hunk 3]
```diff
@@ -477,6 +485,21 @@ under the License.
...
+        "fs.oss.accessKeySecret" = "yyyyyyyyyyyyyyyyyyyy",
```
Comment (wuyunfeng) [inline docs/zh-CN/sql-reference/sql-statements/Data Manipulation/BROKER LOAD.md:499]:
```suggestion
        "fs.oss.secretKey" = "yyyyyyyyyyyyyyyyyyyy",
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java [hunk 4]
```diff
@@ -227,8 +227,8 @@ private static void checkPath(String path) throws AnalysisException {
             URI uri = new URI(path);
             String schema = uri.getScheme();
             if (schema == null || (!schema.equalsIgnoreCase("bos") && !schema.equalsIgnoreCase("afs")
-                    && !schema.equalsIgnoreCase("hdfs"))) {
-                throw new AnalysisException("Invalid export path. please use valid 'HDFS://', 'AFS://' or 'BOS://' path.");
+                    && !schema.equalsIgnoreCase("hdfs") && !schema.equalsIgnoreCase("oss"))) {
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java:230]: If many `fs` or `object storage`, we can use a `enmu` to judge this

**Review by wuyunfeng — CHANGES_REQUESTED** [review]

File: docs/zh-CN/sql-reference/sql-statements/Data Manipulation/BROKER LOAD.md (hunk 2 above)
Comment (gengjun-git) [reply docs/zh-CN/sql-reference/sql-statements/Data Manipulation/BROKER LOAD.md:498]: we use hadoop oss api to access aliyun oss, and the config params is fs.oss.accessKeyId, fs.oss.accessKeySecret. https://developer.aliyun.com/article/292792
And like s3 config, we'd better use the same param to hadoop ecology.

**Comment by zhouaoe** [comment]
@gengjun-git  why didn't submit this  featur？ Is there any problem？ I am making the same  feature，do you will to reopen it again？

**Comment by zhouaoe** [comment]
@gengjun-git Do you will reopen it again？  I am making the same  feature，if you have done, it would  be better that use your code.

File: fs_brokers/apache_hdfs_broker/build.sh (hunk 1 above)
Comment (zhouaoe) [reply fs_brokers/apache_hdfs_broker/build.sh:49]: Hadoop-aliyun(OSS) was merged into hadoop since  hadoop 2.9.1

**Pull request closed without merging**